api/src/main/java/io/serverlessworkflow/api/deserializers/DataInputSchemaDeserializer.java

```diff
@@ -46,17 +53,43 @@ public DataInputSchema deserialize(JsonParser jp, DeserializationContext ctxt)
     JsonNode node = jp.getCodec().readTree(jp);
 
     DataInputSchema dataInputSchema = new DataInputSchema();
+    JsonNode schemaDefinition = null;
 
-    if (!node.isObject()) {
-      dataInputSchema.setSchema(node.asText());
-      dataInputSchema.setFailOnValidationErrors(true); // default
-
-      return dataInputSchema;
-    } else {
-      dataInputSchema.setSchema(node.get("schema").asText());
+    if (node.isObject() && node.get("schema").isObject() && !node.get("schema").isEmpty()) {
+      schemaDefinition = node.get("schema");
       dataInputSchema.setFailOnValidationErrors(node.get("failOnValidationErrors").asBoolean());
+    } else {
+      String schemaFileDef = node.isObject() ? node.get("schema").asText() : node.asText();
+      dataInputSchema.setFailOnValidationErrors(true);
+      dataInputSchema.setRefValue(schemaFileDef);
+      String schemaFileSrc = Utils.getResourceFileAsString(schemaFileDef);
+      JsonNode schemaRefNode;
+      ObjectMapper jsonWriter = new ObjectMapper();
+      if (schemaFileSrc != null && schemaFileSrc.trim().length() > 0) {
+        // if its a yaml def convert to json first
+        if (!schemaFileSrc.trim().startsWith("{")) {
+          // convert yaml to json to validate
+          ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
+          Object obj = yamlReader.readValue(schemaFileSrc, Object.class);
+
+          schemaRefNode =
+              jsonWriter.readTree(new JSONObject(jsonWriter.writeValueAsString(obj)).toString());
+        } else {
+          schemaRefNode = jsonWriter.readTree(new JSONObject(schemaFileSrc).toString());
```

Comment:
Will it lead to case where schemaFileDef class is neither json nor yaml? what would be the behavior we want to exhibit in that case?

Reply:
I don't think we accept a format other than json nor yaml.

Reply:
idk if inline would make sense for yaml defs